### PR TITLE
[Resto Shaman] 3 out of 4 potency conduits

### PR DESCRIPTION
**Pull request by abelito75**
Embrace of Earth, Heavy Rainfall and Swirling Currents

### CODE REVIEW

File: src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+class SwirlingCurrents extends Analyzer {
+
+  healingBoost = 0;
+
+  healing = 0;
+
+  targetsWithBoostedRiptides = [];
+
+  /**
+   * Whenever you cast a Gust of Mist procing ability it reduces the cooldown of Yu'lon or Chi-ji by .5 seconds as well as increasing their healing by x%
```
Comment (niseko) [inline src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js:24]: no monking

File: src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell([SPELLS.HEALING_SURGE_RESTORATION, SPELLS.HEALING_WAVE, SPELLS.RIPTIDE]), this.normalizeBoost);
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.RIPTIDE), this.trackRiptide);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.RIPTIDE), this.removeRiptide);
+    this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.RIPTIDE), this.pandemicRiptide);
+  }
+
+  normalizeBoost(event){
+    if (this.selectedCombatant.hasBuff(SPELLS.SWIRLING_CURRENTS_BUFF.id) || this.targetsWithBoostedRiptides[event.targetID]) {
```
Comment (niseko) [inline src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js:44]: did you check if this correctly catches pwave copied waves, probably will but worth checking as thats a huge swing in hps.

**Review by niseko — CHANGES_REQUESTED** [review]

File: src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+
+  removeRiptide(event){
+    delete this.targetsWithBoostedRiptides[event.targetID];
+  }
+
+  pandemicRiptide(event){
+    if (this.selectedCombatant.hasBuff(SPELLS.SWIRLING_CURRENTS_BUFF.id)) {
+     delete this.targetsWithBoostedRiptides[event.targetID];
```
Comment (niseko) [inline src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js:64]: what does this actually do, when you cast a riptide on a target while you have the swirling buff, it removes it from the tracker? shouldn't that only happen when you don't have the buff

Comment (abelito75) [reply src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js:64]: i forgot a bang

File: src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js (hunk 2 above)
Comment (abelito75) [reply src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js:44]: naw it won't. I fix

File: src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js (hunk 1 above)
Comment (abelito75) [reply src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js:24]: qqMonk

File: src/parser/shaman/restoration/modules/shadowlands/conduits/EmbraceOfEarth.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      const boostedHeal = (event.amount || 0) + (event.absorbed || 0) + (event.overheal || 0);
+      const heal = boostedHeal / (1 + EARTH_SHIELD_BOOST + this.boost);
+      const bonusHeal = heal * this.boost;
+      const effectiveHealing = Math.max(0, (bonusHeal - (event.overheal || 0)));
+      this.healing += effectiveHealing;
+  }
+  }
+
+  statistic() {
+    console.log(this.healing);
```
[the remark this answers is not in the document]
Comment (abelito75) [reply src/parser/shaman/restoration/modules/shadowlands/conduits/EmbraceOfEarth.js:61]: No

**Review by niseko — APPROVED** [review]
@abelito75 
image

**Pull request merged**